### PR TITLE
Fix proto namespace

### PR DESCRIPTION
**Pull request by bddppq**
The package name in the .proto file appears in some string values in the generated .pb.cc and .pb.h files, our define trick doesn't work. We need to replace the package name with the real namespace before invoking protoc.

@yinghai

### CODE REVIEW

File: onnx/gen_proto.py [hunk 1]
```diff
@@ -41,6 +41,14 @@ def process_ifs(lines, onnx_ml):
             elif (2 == in_if and not onnx_ml):
                 yield line
 
+
+PACKAGE_NAME_REGEX = re.compile(r'\{PACKAGE_NAME\}')
+
+def process_package_name(lines, package_name):
```
Comment (yinghai) [inline onnx/gen_proto.py]: Cool, this still happens in setup.py, right? We also need to call this in cmake.

File: CMakeLists.txt [hunk 2]
```diff
@@ -17,14 +17,6 @@ if (WIN32)
 endif()
 add_subdirectory(third_party/pybind11)
 FIND_PACKAGE(Protobuf REQUIRED)
-include_directories(
```
Comment (yinghai) [inline CMakeLists.txt]: Why do we delete this?

Comment (bddppq) [reply CMakeLists.txt]: they are moved to individual libs (e.g. https://github.com/onnx/onnx/blob/a6f7955392ed7f4518c655edd17573ba01bf055a/CMakeLists.txt#L87)

**Comment by bddppq** [comment]
There are two issues with this approach:

1. Our setup.py is doing incremental build, but we don't take environment variable as dependencies, so changing environment variable doesn't trigger a rebuild of the proto files. (This is the reason causing the CI failures right now).
2. We check in the generated proto files into the repo, if we build with a namespace value other than the default one "onnx" it will makes the proto files shown as modified in the working copy. (This one should be abled to solved by changing the gen_proto.py to genereate the proto files outside of the source directory.

**Comment by ezyang** [comment]
In the new patchset, how do I go about updating the checked in generated files?

**Comment by bddppq** [comment]
@ezyang You mean after updating the .in.proto file? As before by running the gen_proto.py. It defaults to use "onnx" as namespace and write to the source directory.

**Comment by ezyang** [comment]
OK, LGTM!

**Comment by bddppq** [comment]
@linkerzhang I'm merging this now since currently our proto files in the repo are ill-formed so it would be better to fix this issue as soon as possible. If you have other review comments, I can make another PR later.

**Pull request merged**